### PR TITLE
Integrate pipeline studio in gitness

### PR DESCRIPTION
**Pull request by srdjan-harness**

### CODE REVIEW

File: apps/gitness/package.json [hunk 1]
```diff
@@ -16,13 +16,22 @@
...
     "@harnessio/playground": "workspace:*",
+    "@harnessio/unified-pipeline": "workspace:*",
+    "@harnessio/yaml-editor": "workspace:*",
     "@tanstack/react-query": "4.36.1",
+    "clsx": "^2.1.1",
+    "monaco-editor": "0.50.0",
     "react": "^18.3.1",
-    "react-dom": "^18.3.1"
+    "react-dom": "^18.3.1",
+    "restful-react": "^15.9.4",
```
Comment (vardanbansal-harness) [inline apps/gitness/package.json:28]: We're not going to use `restful-react` any more. Do we still need it somewhere?

Comment (srdjan-harness) [reply apps/gitness/package.json]: Removed

File: apps/gitness/src/pages/pipeline-edit/components/pipeline-studio-step-form.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+  // const title = currentStepFormDefinition?.identifier
+  // const description = currentStepFormDefinition?.description
+
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  const inputs = (currentStepFormDefinition?.spec as any)?.spec?.inputs as ApiInputs
+
+  const formInputs: IFormDefinition['inputs'] = useMemo(() => {
+    return Object.keys(inputs ?? {}).map(inputName => {
+      return apiInput2IInputDefinition(inputName, inputs[inputName], 'spec.inputs')
```
[the remark this answers is not in the document]
Comment (srdjan-harness) [reply apps/gitness/src/pages/pipeline-edit/components/pipeline-studio-step-form.tsx:45]: Will be included in next PR.

File: apps/gitness/src/App.tsx [hunk 3]
```diff
@@ -57,6 +59,24 @@ export default function App() {
...
           path: 'executions',
           element: <ExecutionsPage />
+        },
+        {
+          path: ':spaceId/:repoId',
+          element: <RepoLayout />,
+          children: [
+            {
+              index: true,
+              element: <>Repos list</>
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/App.tsx:69]: repo list page is already implemented..
but why is it a child of `spaceId/repoId`? that should be the repo details page

**Pull request merged**